### PR TITLE
Use sendPayment instead of sendPaymentSync WIP

### PR DESCRIPTION
**Pull request by dannypaz**

### CODE REVIEW

File: src/lnd-actions/send-payment.js [hunk 1]
```diff
@@ -44,14 +35,26 @@ const SEND_PAYMENT_DEADLINE = 30
...
-    client.sendPaymentSync(paymentOptions, { deadline: deadline(SEND_PAYMENT_DEADLINE) }, (err, res) => {
-      if (err) return reject(err)
-      return resolve(res)
-    })
+    try {
+      const call = client.router.sendPayment(paymentOptions)
+
+      call.on('data', data => {
+        if (data.paymentError) {
+          return reject(data.paymentError)
```
Comment (treygriffith) [inline src/lnd-actions/send-payment.js]: probably worth adding a `logger.error` here too

File: src/lnd-actions/send-payment.js [hunk 2]
```diff
@@ -44,14 +35,26 @@ const SEND_PAYMENT_DEADLINE = 30
...
  */
-function sendPayment (paymentOptions, { client }) {
+function sendPayment (paymentOptions, { client, logger }) {
   return new Promise((resolve, reject) => {
-    client.sendPaymentSync(paymentOptions, { deadline: deadline(SEND_PAYMENT_DEADLINE) }, (err, res) => {
-      if (err) return reject(err)
-      return resolve(res)
-    })
+    try {
+      const call = client.router.sendPayment(paymentOptions)
```
Comment (treygriffith) [inline src/lnd-actions/send-payment.js]: do we also need to handle `call.on('error', ...)` and `call.on('cancel', ...)` ?

**Comment by treygriffith** [comment]
[Sparkswap paying invoices leads to stuck HTLCs](https://trello.com/c/grgvEmSy/711-sparkswap-paying-invoices-leads-to-stuck-htlcs)

**Comment by dannypaz** [comment]
closing as this is no longer needed

**Pull request closed without merging**